### PR TITLE
WIP: topics for rest library docs

### PR DESCRIPTION
**Pull request by jskeet**
This is just a starting point for comments on topics I've no doubt missed.

It includes *most* (but not deliberately all) of what's in https://github.com/googleapis/google-api-dotnet-client/blob/main/FAQ.md - once this is all published, we should update that.

### CODE REVIEW

File: docs/devsite-help/legacy-rest-libraries.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Legacy HTTP/1.1 + REST libraries
```
Comment (amanda-tarafa) [inline docs/devsite-help/legacy-rest-libraries.md:1]: If I were to use YouTube or Drive, I would be slightly scared of the legacy word here. Not certain how to address that, but maybe simply aknowledging that these are the only libraries for such APIs is fine.

**Review by amanda-tarafa — COMMENTED** [review]
Yep, this looks good

File: docs/devsite-help/legacy-rest-libraries.md (hunk 1 above)
Comment (jskeet) [reply docs/devsite-help/legacy-rest-libraries.md:1]: That's a good point. I think we can still keep the term in the library types doc, but I'll remove it from here. I can mention that these are effectively legacy for GCP.

**Comment by jskeet** [comment]
Closing as a PR now, as it's far from ready for review.

**Pull request closed without merging**